File: tests/snapshot_pre_upgrade/fox.java
import roundtrip.builtins_py.*;

import java.util.*;



public class Box {
    public Box(Object value) {
        this.history_len = 0;
    }
    public Object update_value(Object new_value) {
        this.value = new_value;
        this.history_len = (this.history_len + 1);
    }
    public Object describe() {
        return (((("Box(value=" + builtins_py.str(this.value)) + ", writes=") + builtins_py.str(this.history_len)) + ")");
    }
}

public class Main {
public static void side_effect_marker(Object tag) {
    builtins_py.print(("SIDE_EFFECT:" + tag));
    return true;
}
public static void short_circuit_and_demo() {
    Object result = (false && side_effect_marker("AND_RHS"));
    builtins_py.print(("and_short_circuit_result=" + bool_str(result)));
}
public static void short_circuit_or_demo() {
    Object result = (true || side_effect_marker("OR_RHS"));
    builtins_py.print(("or_short_circuit_result=" + bool_str(result)));
}
public static void bool_str(Object b) {
    if (b) {
        return "True";
    } else {
        return "False";
    }
}
public static void none_str(Object v) {
    if ((v == null)) {
        return "None";
    }
    return builtins_py.str(v);
}
public static void classify_number(Object n) {
    if ((n < 0)) {
        return "negative";
    } else if ((n == 0)) {
        return "zero";
    } else if ((n < 10)) {
        return "small";
    } else {
        return "large";
    }
}
public static void factorial_recursive(Object n) {
    if ((n <= 1)) {
        return 1;
    }
    return (n * factorial_recursive((n - 1)));
}
public static void risky_divide(Object a, Object b) {
    if ((b == 0)) {
        throw new IllegalArgumentException("division by zero not allowed");
    }
    return (a / b);
}
public static void main() {
    Object a = 17;
    Object b = 5;
    Object add_i = (a + b);
    Object sub_i = (a - b);
    Object mul_i = (a * b);
    Object div_f = (a / b);
    Object div_i = ((int)(a / b));
    Object mod_i = (a % b);
    Object neg_i = -(a);
    builtins_py.print(("add_i=" + builtins_py.str(add_i)));
    builtins_py.print(("sub_i=" + builtins_py.str(sub_i)));
    builtins_py.print(("mul_i=" + builtins_py.str(mul_i)));
    builtins_py.print(("div_f=" + builtins_py.str(div_f)));
    builtins_py.print(("div_i=" + builtins_py.str(div_i)));
    builtins_py.print(("mod_i=" + builtins_py.str(mod_i)));
    builtins_py.print(("neg_i=" + builtins_py.str(neg_i)));
    Object fa = 9.0;
    Object fb = 2.0;
    builtins_py.print(("float_div=" + builtins_py.str((fa / fb))));
    builtins_py.print(("float_floordiv=" + builtins_py.str(((int)(fa / fb)))));
    builtins_py.print(("eq=" + bool_str((a == b))));
    builtins_py.print(("ne=" + bool_str((a != b))));
    builtins_py.print(("lt=" + bool_str((a < b))));
    builtins_py.print(("le=" + bool_str((a <= b))));
    builtins_py.print(("gt=" + bool_str((a > b))));
    builtins_py.print(("ge=" + bool_str((a >= b))));
    short_circuit_and_demo();
    short_circuit_or_demo();
    builtins_py.print(("not_demo=" + bool_str(!((a == b)))));
    for (Object n : new ArrayList<>(Arrays.asList(-(3), 0, 4, 42))) {
        builtins_py.print(((("classify(" + builtins_py.str(n)) + ")=") + classify_number(n)));
    }
    Object i = 0;
    Object total = 0;
    while ((i < 5)) {
        total = (total + i);
        i = (i + 1);
    }
    builtins_py.print(("while_total=" + builtins_py.str(total)));
    Object range_sum = 0;
    for (Object k : builtins_py.range(1, 6)) {
        range_sum = (range_sum + k);
    }
    builtins_py.print(("range_sum=" + builtins_py.str(range_sum)));
    Object words = new ArrayList<>(Arrays.asList("quick", "brown", "fox"));
    Object joined = "";
    for (Object w : words) {
        joined = ((joined + w) + "-");
    }
    builtins_py.print(("joined=" + joined));
    builtins_py.print(("factorial5=" + builtins_py.str(factorial_recursive(5))));
    Object counter = 0;
    counter = (counter + 1);
    counter = (counter + 1);
    builtins_py.print(("counter=" + builtins_py.str(counter)));
    Object fox_list = new ArrayList<>(Arrays.asList(1, 2, 3));
    fox_list[0] = 100;
    fox_list.append(4);
    builtins_py.print(("fox_list0=" + builtins_py.str(fox_list[0])));
    builtins_py.print(("fox_list_len=" + builtins_py.str(builtins_py.len(fox_list))));
    builtins_py.print(("fox_list_last=" + builtins_py.str(fox_list[3])));
    Object fox_map = new HashMap<>();
    fox_map["quick"] = 1;
    fox_map["brown"] = 2;
    fox_map["fox"] = 3;
    builtins_py.print(("map_get_brown=" + builtins_py.str(fox_map.get("brown"))));
    fox_map["fox"] = 30;
    builtins_py.print(("map_get_fox=" + builtins_py.str(fox_map.get("fox"))));
    Object name = "fox";
    Object speed = 12;
    Object formatted = "The {} runs at {} mph".format(name, speed);
    builtins_py.print(formatted);
    builtins_py.print(("name_len=" + builtins_py.str(builtins_py.len(name))));
    Object pair = new ArrayList<>(Arrays.asList(name, speed));
    Object p_name = pair;
    builtins_py.print(((("pair_name=" + p_name) + " pair_speed=") + builtins_py.str(p_speed)));
    Object box = Box(10);
    builtins_py.print(("box_before=" + box.describe()));
    box.update_value(99);
    builtins_py.print(("box_after=" + box.describe()));
    builtins_py.print(("box_value=" + builtins_py.str(box.value)));
    try {
        risky_divide(10, 0);
    } catch (Exception e) {
    }
    Object safe_result = risky_divide(10, 4);
    builtins_py.print(("safe_result=" + builtins_py.str(safe_result)));
    Object maybe_value = null;
    builtins_py.print(("maybe_before=" + none_str(maybe_value)));
    maybe_value = 7;
    builtins_py.print(("maybe_after=" + none_str(maybe_value)));
    builtins_py.print(("contains_fox=" + bool_str((words.contains("fox")))));
    builtins_py.print(("contains_wolf=" + bool_str((words.contains("wolf")))));
    builtins_py.print(("contains_key=" + bool_str((fox_map.contains("quick")))));
    builtins_py.print("DONE");
}
    public static void main(String[] args) {
        main();
    }
}

File: tests/snapshot_pre_upgrade/fox.kt
import roundtrip.builtins_py.*

fun side_effect_marker(tag: Any?): Unit {
    print(("SIDE_EFFECT:" + tag))
    return true
}
fun short_circuit_and_demo(): Unit {
    var result: Any? = (false && side_effect_marker("AND_RHS"))
    print(("and_short_circuit_result=" + bool_str(result)))
}
fun short_circuit_or_demo(): Unit {
    var result: Any? = (true || side_effect_marker("OR_RHS"))
    print(("or_short_circuit_result=" + bool_str(result)))
}
fun bool_str(b: Any?): Unit {
    if (b) {
        return "True"
    } else {
        return "False"
    }
}
fun none_str(v: Any?): Unit {
    if ((v === null)) {
        return "None"
    }
    return str(v)
}
fun classify_number(n: Any?): Unit {
    if ((n < 0)) {
        return "negative"
    } else if ((n == 0)) {
        return "zero"
    } else if ((n < 10)) {
        return "small"
    } else {
        return "large"
    }
}
fun factorial_recursive(n: Any?): Unit {
    if ((n <= 1)) {
        return 1
    }
    return (n * factorial_recursive((n - 1)))
}
class Box {
    constructor(value: Any?) {
        this.history_len = 0
    }
    fun update_value(new_value: Any?): Unit {
        this.value = new_value
        this.history_len = (this.history_len + 1)
    }
    fun describe(): Unit {
        return (((("Box(value=" + str(this.value)) + ", writes=") + str(this.history_len)) + ")")
    }
}
fun risky_divide(a: Any?, b: Any?): Unit {
    if ((b == 0)) {
        throw IllegalArgumentException("division by zero not allowed")
    }
    return (a / b)
}
fun main(): Unit {
    var a: Any? = 17
    var b: Any? = 5
    var add_i: Any? = (a + b)
    var sub_i: Any? = (a - b)
    var mul_i: Any? = (a * b)
    var div_f: Any? = (a / b)
    var div_i: Any? = (a / b).toInt()
    var mod_i: Any? = (a % b)
    var neg_i: Any? = -(a)
    print(("add_i=" + str(add_i)))
    print(("sub_i=" + str(sub_i)))
    print(("mul_i=" + str(mul_i)))
    print(("div_f=" + str(div_f)))
    print(("div_i=" + str(div_i)))
    print(("mod_i=" + str(mod_i)))
    print(("neg_i=" + str(neg_i)))
    var fa: Any? = 9.0
    var fb: Any? = 2.0
    print(("float_div=" + str((fa / fb))))
    print(("float_floordiv=" + str((fa / fb).toInt())))
    print(("eq=" + bool_str((a == b))))
    print(("ne=" + bool_str((a != b))))
    print(("lt=" + bool_str((a < b))))
    print(("le=" + bool_str((a <= b))))
    print(("gt=" + bool_str((a > b))))
    print(("ge=" + bool_str((a >= b))))
    short_circuit_and_demo()
    short_circuit_or_demo()
    print(("not_demo=" + bool_str(!((a == b)))))
    for (n in mutableListOf(-(3), 0, 4, 42)) {
        print(((("classify(" + str(n)) + ")=") + classify_number(n)))
    }
    var i: Any? = 0
    var total: Any? = 0
    while ((i < 5)) {
        total = (total + i)
        i = (i + 1)
    }
    print(("while_total=" + str(total)))
    var range_sum: Any? = 0
    for (k in (1 until 6)) {
        range_sum = (range_sum + k)
    }
    print(("range_sum=" + str(range_sum)))
    var words: Any? = mutableListOf("quick", "brown", "fox")
    var joined: Any? = ""
    for (w in words) {
        joined = ((joined + w) + "-")
    }
    print(("joined=" + joined))
    print(("factorial5=" + str(factorial_recursive(5))))
    var counter: Any? = 0
    counter = (counter + 1)
    counter = (counter + 1)
    print(("counter=" + str(counter)))
    var fox_list: Any? = mutableListOf(1, 2, 3)
    fox_list[0] = 100
    fox_list.append(4)
    print(("fox_list0=" + str(fox_list[0])))
    print(("fox_list_len=" + str(len(fox_list))))
    print(("fox_list_last=" + str(fox_list[3])))
    var fox_map: Any? = mutableMapOf()
    fox_map["quick"] = 1
    fox_map["brown"] = 2
    fox_map["fox"] = 3
    print(("map_get_brown=" + str(fox_map.get("brown"))))
    fox_map["fox"] = 30
    print(("map_get_fox=" + str(fox_map.get("fox"))))
    var name: Any? = "fox"
    var speed: Any? = 12
    var formatted: Any? = "The {} runs at {} mph".format(name, speed)
    print(formatted)
    print(("name_len=" + str(len(name))))
    var pair: Any? = mutableListOf(name, speed)
    var p_name: Any? = pair
    print(((("pair_name=" + p_name) + " pair_speed=") + str(p_speed)))
    var box: Any? = Box(10)
    print(("box_before=" + box.describe()))
    box.update_value(99)
    print(("box_after=" + box.describe()))
    print(("box_value=" + str(box.value)))
    try {
        risky_divide(10, 0)
    } catch (e: Exception) {
    }
    var safe_result: Any? = risky_divide(10, 4)
    print(("safe_result=" + str(safe_result)))
    var maybe_value: Any? = null
    print(("maybe_before=" + none_str(maybe_value)))
    maybe_value = 7
    print(("maybe_after=" + none_str(maybe_value)))
    print(("contains_fox=" + bool_str(("fox" in words))))
    print(("contains_wolf=" + bool_str(("wolf" in words))))
    print(("contains_key=" + bool_str(("quick" in fox_map))))
    print("DONE")
}

fun main(args: Array<String>) {
    main()
}